skimage/io/collection.py

```diff
@@ -256,10 +257,15 @@ def __getitem__(self, n):
                 if self._frame_index:
                     fname, img_num = self._frame_index[n]
                     if img_num is not None:
-                        self.data[idx] = self.load_func(fname, img_num=img_num,
-                                                        **kwargs)
-                    else:
+                        kwargs['img_num'] = img_num
```

Comment:
This is all a bit messy... can we use Python's inspections here instead?

```python
import inspect
params = inspect.signature(load_func).parameters
if img_num is not None and 'img_num' in params.keys():
    kwargs['img_num'] = img_num
```

optionally add: `or any(p.kind == p.VAR_KEYWORD for p in params.values())`